src/components/NavBar/store.js

```diff
@@ -1,20 +1,80 @@
-import { observable, action } from 'mobx';
+import { observable, action, computed } from 'mobx';
+import { OracleStatus, Token } from 'constants';
+import _ from 'lodash';
+import Topic from '../../stores/models/Topic';
+import Oracle from '../../stores/models/Oracle';
+
+import { queryAllTopics, queryAllOracles, queryAllVotes } from '../../network/graphQuery';
 
 const INIT_VALUES = {
   myActivitesCount: 0, // INIT_VALUESial loaded state
 };
 
 
-export default class NavBarStore {
-  @observable myActivitesCount = INIT_VALUES.myActivitesCount
+export default class {
+  @observable ResultSettingCount = INIT_VALUES.ResultSettingCount
+  @observable FinalizeCount = INIT_VALUES.FinalizeCount
+  @observable WithdrawCount = INIT_VALUES.WithdrawCount
+  @computed get totalCount() {
+    return this.ResultSettingCount + this.FinalizeCount + this.WithdrawCount;
+  }
 
-  @action.bound
-  async init() {
-    // this.myActivitesCount = await this.getMyActivitiesCount();
+  constructor(app) {
+    this.app = app;
   }
 
-  // @action.bound
-  // async getMyActivitiesCount() {
+  @action
+  init() {
+    this.getCount();
+  }
+
+  @action.bound
+  async getCount() {
+    try {
+      const voteFilters = [];
+      const topicFilters = [];
 
-  // }
+      // Get all votes for all your addresses
+      _.each(this.app.wallet.addresses, (item) => {
+        voteFilters.push({ voterQAddress: item.address });
+        topicFilters.push({ status: OracleStatus.WITHDRAW, creatorAddress: item.address });
+      });
+
+      // Filter votes
+      let votes = await queryAllVotes(voteFilters);
+      votes = votes.reduce((accumulator, vote) => {
+        const { voterQAddress, topicAddress, optionIdx } = vote;
+        if (!_.find(accumulator, { voterQAddress, topicAddress, optionIdx })) accumulator.push(vote);
+        return accumulator;
+      }, []);
+
+      _.each(votes, ({ topicAddress, optionIdx }) => {
+        topicFilters.push({ status: OracleStatus.WITHDRAW, address: topicAddress, resultIdx: optionIdx });
+      });
+      let topics = await queryAllTopics(topicFilters);
+      topics = _.uniqBy(topics, 'txid').map((topic) => new Topic(topic, this.app));
+      this.WithdrawCount = topics.length;
+
+      // Get result set items
+      const oracleSetFilters = [{ token: Token.QTUM, status: OracleStatus.OPEN_RESULT_SET }];
+      _.each(action.walletAddresses, (item) => {
+        oracleSetFilters.push({
+          token: Token.QTUM,
+          status: OracleStatus.WAIT_RESULT,
+          resultSetterQAddress: item.address,
+        });
+      });
+      let oracles = await queryAllOracles(oracleSetFilters);
+      oracles = _.uniqBy(oracles, 'txid').map((oracle) => new Oracle(oracle, this.app));
```

Comment:
Same comment as above. `uniqBy` not necessary. If there is duplicates, we should see the error so we can fix it instead of failing silently. This can lead to harder debugging later, or we would never know something is wrong with the BE.

No need to map to Oracle, we are only using the length.

Reply:
if it's not necessary, why do we do it for rendering the cards and why we have them in the old saga?

Reply:
That is a great question. I believe they were just there before. Remove and if there are problems, we can tackle them then

Reply:
Let's remove all the uniqBy and map to Oracles since they are not used.